rust/tensors.rs

```diff
@@ -118,6 +126,37 @@ impl<'a> TensorDescriptor<'a> {
     }
 }
 
+impl fmt::Display for ElementType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let element_kind = String::from(match self {
+            ElementType::Bool => "b",
+            ElementType::UInt8 => "u8",
+            ElementType::Int8 => "i8",
+            ElementType::Int16 => "i16",
+            ElementType::Int32 => "i32",
+            ElementType::Int64 => "i64",
+            ElementType::Float16 => "f16",
+            ElementType::Float32 => "f32",
+            ElementType::Float64 => "f64",
+            ElementType::Complex64 => "c64",
+            ElementType::Complex128 => "c128",
+            ElementType::String => "string",
+            _ => "?"
+        });
+
+        write!(f, "{}", element_kind)
```

Comment:
We shouldn't need to allocate a new `String` here because we are just using literals.

```suggestion
        let element_kind = match self {
            ElementType::Bool => "b",
            ElementType::UInt8 => "u8",
            ElementType::Int8 => "i8",
            ElementType::Int16 => "i16",
            ElementType::Int32 => "i32",
            ElementType::Int64 => "i64",
            ElementType::Float16 => "f16",
            ElementType::Float32 => "f32",
            ElementType::Float64 => "f64",
            ElementType::Complex64 => "c64",
            ElementType::Complex128 => "c128",
            ElementType::String => "string",
            _ => "?"
        };

        f.write_str(element_kind)
```